From stable-bounces@linux.kernel.org Mon Dec 12 12:27:13 2005
Date: Thu, 24 Nov 2005 00:00:03 -0800
Message-Id: <200511240800.jAO803jw027240@hera.kernel.org>
From: Linux Kernel Mailing List <linux-kernel@vger.kernel.org>
To: git-commits-head@vger.kernel.org
Cc: 
Subject: BRIDGE: recompute features when adding a new device

From: Olaf Rempel <razzor@kopf-tisch.de>

[BRIDGE]: recompute features when adding a new device

We must recompute bridge features everytime the list of underlying
devices changes, or we might end up with features that are not supported
by all devices (eg. NETIF_F_TSO)
This patch adds the missing recompute when adding a device to the bridge.

Signed-off-by: Olaf Rempel <razzor@kopf-tisch.de>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 net/bridge/br_if.c |    1 +
 1 file changed, 1 insertion(+)

--- linux-2.6.14.3.orig/net/bridge/br_if.c
+++ linux-2.6.14.3/net/bridge/br_if.c
@@ -366,6 +366,7 @@ int br_add_if(struct net_bridge *br, str
 
 		spin_lock_bh(&br->lock);
 		br_stp_recalculate_bridge_id(br);
+		br_features_recompute(br);
 		if ((br->dev->flags & IFF_UP) 
 		    && (dev->flags & IFF_UP) && netif_carrier_ok(dev))
 			br_stp_enable_port(p);
